nistats/glm.py

```diff
@@ -58,183 +47,394 @@ def data_scaling(Y):
-            behaviour), the whole vector is returned
+        imgs: Niimg-like object or list of Niimg-like objects,
+            See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+            Data on which the GLM will be fitted. If this is a list,
+            the affine is considered the same for all.
 
-        Returns
-        -------
-        beta: array of shape (n_voxels, n_columns)
-            the required coefficients
-        """
-        # make colum_index a list if it an int
-        if column_index is None:
-            column_index = np.arange(self.X.shape[1])
-        if not hasattr(column_index, '__iter__'):
-            column_index = [int(column_index)]
-        n_beta = len(column_index)
-
-        # build the beta array
-        beta = np.zeros((n_beta, self.labels_.size), dtype=np.float)
-        for l in self.results_.keys():
-            beta[:, self.labels_ == l] = self.results_[l].theta[column_index]
-        return beta
-
-    def get_mse(self):
-        """Acessor for the mean squared error of the model
+        design_matrices: array or list of arrays
+            Models of the temporal events considered in the analysis.
+            the arrays have shape (n_time_points, n_regressors)
 
-        Returns
-        -------
-        mse: array of shape (n_voxels)
-            the sum of square error per voxel
         """
-        # build the beta array
-        mse = np.zeros(self.labels_.size, dtype=np.float)
-        for l in self.results_.keys():
-            mse[self.labels_ == l] = self.results_[l].MSE
-        return mse
+        # First, learn the mask
+        if not isinstance(self.mask, NiftiMasker):
+            self.masker_ = NiftiMasker(
+                mask_img=self.mask, smoothing_fwhm=self.smoothing_fwhm,
+                target_affine=self.target_affine,
+                standardize=self.standardize, low_pass=self.low_pass,
+                high_pass=self.high_pass, mask_strategy='epi',
+                mask_args={'opening': 0}, t_r=self.t_r, memory=self.memory,
+                verbose=max(0, self.verbose - 1),
+                target_shape=self.target_shape,
+                memory_level=self.memory_level)
+        else:
+            self.masker_ = clone(self.mask)
+            for param_name in ['target_affine', 'target_shape',
+                               'smoothing_fwhm', 'low_pass', 'high_pass',
+                               't_r', 'memory', 'memory_level']:
+                our_param = getattr(self, param_name)
+                if our_param is None:
+                    continue
+                if getattr(self.masker_, param_name) is not None:
+                    warn('Parameter %s of the masker overriden' % param_name)
+                setattr(self.masker_, param_name, our_param)
+
+        # make design_matrices a list of arrays
+        design_matrices_ = [X for X in design_matrices]
+        if isinstance(design_matrices, (_basestring, np.ndarray)):
```

Comment:
This check is a bit weird since you just defined design_matrices to be a list in the previous line...